plots/electricity_bills.py

```diff
@@ -0,0 +1,255 @@
+    n.loads_t.p_set =  n.loads_t.p_set.rename(columns=n.loads.bus.to_dict())
+    
+    # low voltage load
+    lv_buses = n.buses.query("carrier == 'low voltage'").index
+    lv_load = n.loads_t.p_set[lv_buses]
+    
+    # electricity consumption of residential heat techs in low_voltage bus of links
+    rh_elec_links = n.links.query("carrier in @rh_techs_elec").index
+    rh_techs_mapping = n.links.loc[rh_elec_links, "bus0"].to_dict()
+    rh_techs_consume = n.links_t.p0[rh_elec_links]
+    rh_techs_consume = rh_techs_consume.rename(columns=rh_techs_mapping)
+    rh_techs_consume = rh_techs_consume.groupby(level=0, axis=1).sum()
+    
+    # electricity consumption of land transport EV in low_voltage bus in links
+    ev_charge_links = n.links.query("carrier in @ev_tech_charge").index
+    ev_charge_mapping = n.links.loc[ev_charge_links, "bus0"].to_dict()
+    ev_tech_consume = n.links_t.p0[ev_charge_links]
+    ev_tech_consume = ev_tech_consume.rename(columns=ev_charge_mapping)
+    ev_tech_consume = ev_tech_consume.groupby(level=0, axis=1).sum()
+    
+    # electricity prosumption of land transport EV to low_voltage bus in links
+    ev_discharge_links = n.links.query("carrier in @ev_tech_discharge").index
+    ev_discharge_mapping = n.links.loc[ev_discharge_links, "bus1"].to_dict()
+    ev_tech_prosume = n.links_t.p1[ev_discharge_links]
+    ev_tech_prosume = ev_tech_prosume.rename(columns=ev_discharge_mapping)
+    ev_tech_prosume = ev_tech_prosume.groupby(level=0, axis=1).sum()
+    
+    # electricity prosumption of micro CHP to low_voltage bus in links
+    rh_mCHP_links = n.links.query("carrier in @rh_techs_mCHP").index
+    rh_mCHP_lv_mapping = n.links.loc[rh_mCHP_links, "bus1"].to_dict()
+    rh_mCHP_prosume = n.links_t.p1[rh_mCHP_links]
+    rh_mCHP_prosume = rh_mCHP_prosume.rename(columns=rh_mCHP_lv_mapping)
+    rh_mCHP_prosume = rh_mCHP_prosume.groupby(level=0, axis=1).sum()
+    
+    # gas consumption of micro CHP in EU_gas bus in links
+    rh_mCHP_gas_mapping = n.links.loc[rh_mCHP_links, "bus0"].to_dict()
+    rh_mCHP_consume = n.links_t.p0[rh_mCHP_links]
+    rh_mCHP_consume = rh_mCHP_consume.rename(columns=rh_mCHP_gas_mapping)
+    rh_mCHP_consume = rh_mCHP_consume.groupby(level=0, axis=1).sum()
+    
+    # gas consumption of gas boilers in EU_gas bus in links
+    rh_gas_links = n.links.query("carrier in @rh_techs_gas").index
+    if not rh_gas_links.empty:   
+        rh_gas_mapping = n.links.loc[rh_gas_links, "bus0"].to_dict()
+        rh_gas_consume = n.links_t.p0[rh_gas_links]
+        rh_gas_consume = rh_gas_consume.rename(columns=rh_gas_mapping)
+        rh_gas_consume = rh_gas_consume.groupby(level=0, axis=1).sum()
+    else:
+        rh_gas_consume = pd.DataFrame()
+    
+    # total electricity consumption in low_voltage nodes
+    total_elec_load = (lv_load + rh_techs_consume + ev_tech_consume + ev_tech_prosume + rh_mCHP_prosume).multiply(n.snapshot_weightings.stores, axis=0)
```

Comment:
What about `rh_mCHP_consume`?

```suggestion
    total_elec_load = (lv_load + rh_techs_consume + ev_tech_consume + ev_tech_prosume + rh_mCHP_consume + rh_mCHP_prosume).multiply(n.snapshot_weightings.stores, axis=0)
```

Reply:
> In mCHP, we have gas as consumable in bus0, while electricity (bus1) and heat (bus2) as product. `rh_mCHP_consume` is considered in gas part, we only have prosume here in electricity.

Reply:
I wonder why in urban central, bus1 is not the low voltage bus...

Reply:
> Agree, thanks for clarification!
> 
> I wonder why in urban central, bus1 is not the low voltage bus... I think this is an error in upstream pypsa-eur?

Do you mean `urban central solid biomass CHP`? For solid biomass micro CHPs in residential and services, bus1 is low_voltage. But for urban central we do not have mCHP, instead we have regular `solid biomass CHP` which is connected to grid (i.e. bus1 is AC bus).

Reply:
No I think I saw it for gas boilers, for example, or some of the heat pumps, but I don't remember